business_requirement_deliverable_cost/models/business.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class BusinessRequirementResource(models.Model):
+    _inherit = "business.requirement.resource"
+
+    sale_price_unit = fields.Float(
+        string='Sales Price',
+        groups='business_requirement_deliverable_cost.'
+        'group_business_requirement_estimation',
+    )
+    sale_price_total = fields.Float(
+        compute='_compute_sale_price_total',
+        string='Total Revenue',
+        groups='business_requirement_deliverable_cost.'
+        'group_business_requirement_estimation',
+    )
+    unit_price = fields.Float(
+        string='Cost Price',
+        groups='business_requirement_deliverable_cost.'
+        'group_business_requirement_cost_control',
+    )
+    price_total = fields.Float(
+        store=False,
+        compute='_compute_get_price_total',
+        string='Total Cost',
+        groups='business_requirement_deliverable_cost.'
+        'group_business_requirement_cost_control',
+    )
+
+    @api.multi
+    @api.depends('unit_price', 'qty')
+    def _compute_get_price_total(self):
+        for resource in self:
+            resource.price_total = resource.unit_price * resource.qty
+
+    @api.multi
+    @api.depends('sale_price_unit', 'qty')
+    def _compute_sale_price_total(self):
+        for resource in self:
+            resource.sale_price_total = resource.sale_price_unit * resource.qty
+
+    @api.multi
+    def _get_partner(self):
+        for resource in self:
+            br_id = False
+            if resource.business_requirement_deliverable_id:
+                br_deliverable = resource.business_requirement_deliverable_id
+            if br_deliverable and br_deliverable.business_requirement_id:
+                br_id = br_deliverable.business_requirement_id
+            if br_id and br_id.partner_id:
+                return br_id.partner_id
+            else:
+                return False
+
+    @api.multi
+    def _get_pricelist(self):
+        for resource in self:
```

Comment:
> pedrobaeza 9 days ago
 This should have a self.ensure_one() instead the loop

ok

> dreispt 3 days ago
A computed field can't be overridden manually; a onchange proposed value can.
IMO it's more flexible to keep this defined in an onchange.


Then I'll follow @dreispt recommendations. I also prefer flexibility.